build/Rulesets/Roslyn_BuildRules.ruleset

```diff
@@ -106,6 +106,13 @@
   <Rules AnalyzerId="Microsoft.NetCore.Analyzers" RuleNamespace="Microsoft.NetCore.Analyzers.ImmutableCollections">
     <Rule Id="RS0012" Action="Warning" />
   </Rules>
+  <Rules AnalyzerId="Microsoft.VisualStudio.Threading.Analyzers" RuleNamespace="Microsoft.VisualStudio.Threading.Analyzers">
+    <Rule Id="VSTHRD001" Action="None" /> <!-- don't enforce useage of JTF -->
+    <Rule Id="VSTHRD002" Action="None" /> <!-- don't enforce useage of JTF -->
+    <Rule Id="VSTHRD010" Action="None" /> <!-- don't enforce useage of JTF -->
+    <Rule Id="VSTHRD103" Action="None" /> <!-- without flow analysis this rule has a lot of false positives and provides little value -->
```

Comment:
❔  Was the problem here flow analysis, or lack of support for local functions?

Reply:
I can investigate the implementation further but fundamentally I have never seen `VSTHRD103` catch a product bug.  It uses string matching to try and determine if methods have async versions.  You can look at the source here: https://github.com/Microsoft/vs-threading/blob/29d8c6ebb3758ccf1aa1488feb7083ae4744a267/src/Microsoft.VisualStudio.Threading.Analyzers/VSTHRD103UseAsyncOptionAnalyzer.cs

Reply:
It caught several bugs in GitExtensions during its conversion. Arguably one of the most helpful analyzers of the entire set there.

You indicated in this comment that the analyzer was reporting false positives. I have not seen this occur, and was wondering what you saw. Given how beneficial the analyzer is, I'd like to see bugs filed so they can be fixed.

Reply:
Well in project system we implement or use methods that are named similar to other a sync methods.  As a result we had hundreds of false positives about a synchronous API *that is implementing and interface* needing to call an async method when that is not the method contract, there was a separate async method that *did* call the async api.  Turning this on for roslyn just seems to trigger on `t.Result` calls though, so I have no problem turning it on.